modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestTransactionHelper.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+
+package org.nuxeo.ecm.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+import org.nuxeo.runtime.transaction.TransactionRuntimeException;
+
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeFeature.class, TransactionalFeature.class })
+public class TestTransactionHelper {
+
+    @Test
+    public void testTransactionTtl() throws InterruptedException {
+        TransactionHelper.commitOrRollbackTransaction();
+        int ttl = TransactionHelper.getTransactionTimeToLive();
+        // outside of a transaction
+        assertEquals(-1, ttl);
+        int timeout = 2;
+        TransactionHelper.startTransaction(timeout);
+        ttl = TransactionHelper.getTransactionTimeToLive();
+        // alive
+        assertTrue("Got " + ttl, ttl > 0);
+        // timeout
+        Thread.sleep((timeout + 2) * 1000);
```

Comment:
An option to avoid the wait is to go full whitebox-testing mode and go patch Geronimo through reflection to force the transaction timeout.